pyke/lightcurve.py

```diff
@@ -115,6 +121,107 @@ def fold(self, period, phase=0.):
+             sigma_clip=5., norm_factor=1.4):
+        """Estimate the CDPP noise metric using the Savitzky-Golay (SG) method.
+
+        An interesting estimate of the noise in a lightcurve is the scatter that
+        remains after all long term trends have been removed. This is the kernel
+        of the idea behind the Combined Differential Photometric Precision (CDPP).
+        The Kepler Pipeline uses a wavelet-based algorithm to calculate the
+        signal-to-noise of the specific waveform of transits of various
+        durations. In this implementation, we use the simpler CDPP proxy
+        algorithm as discussed by Gilliland et al (2011ApJS..197....6G)
+        and Van Cleve et al (2016PASP..128g5002V).
+
+        The steps are:
+            1. Remove low frequency signals using a Savitzky-Golay filter.
+            2. Remove outliers using sigma-clipping.
+            3. Compute the standard deviation of a running mean with
+               configurable window length equal to `transit_duration`.
+
+        Parameters
+        ----------
+        transit_duration : int, optional
+            The transit duration in cadences. This is the length of the window
+            used to compute the running mean. The default is 13, which
+            corresponds to a 6.5 hour transit in data sampled at 30-min cadence.
+        savgol_window : int, optional
+            Width of Savitsky-Golay filter in cadences (odd number).
+            Default value 101 (2.0 days in Kepler Long Cadence mode).
+        savgol_polyorder : int, optional
+            Polynomial order of the Savitsky-Golay filter.
+            The recommended value is 2.
+        sigma_clip : float, optional
+            The number of standard deviations to use for clipping outliers.
+            The default is 5.
+        norm_factor : float, optional
+            Noise normalization factor for the SG filter.  Recommended
+            value for 13 cadence transit and 101 cadence window is 1.40.
+
+        Returns
+        -------
+        cdpp : float
+            Savitzky-Golay CDPP noise metric in units parts-per-million (ppm).
+
+        Notes
+        -----
+        This implementation is adapted from the Matlab version used by
+        Jeff van Cleve:
+        svn+ssh://murzim/repo/so/trunk/Develop/jvc/common/compute_SG_noise.m
+        """
+        if not isinstance(transit_duration, int):
+            raise TypeError("transit_duration must be an integer")
+        detrended_lc, _ = self.flatten(window_length=savgol_window,
+                                       polyorder=savgol_polyorder)
+        cleaned_lc = detrended_lc.remove_outliers(sigma=sigma_clip)
+        mean_filter = np.ones(transit_duration) / float(transit_duration)
+        running_mean = np.convolve(cleaned_lc.flux, mean_filter, mode='same')
+        cdpp_ppm = norm_factor * np.std(running_mean) * 1e6
+        return cdpp_ppm
+
```

Comment:
This looks great!
Should we use running mean or running median? If the later, see
https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.medfilt.html

Reply:
The papers mentioned in the docstring use the mean, so that's what we should use.  I guess it makes sense because the median would jump around too much if the window size is small!

Reply:
Cool! ``numpy.cumsum`` looks to be faster than ``numpy.convolve``, see stackoverflow: https://stackoverflow.com/questions/13728392/moving-average-or-running-mean